standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1046,6 +1030,20 @@ private List<Partition> getPartitionsByNames(String catName, String dbName,
     return getPartitionsByQuery(catName, dbName, tblName, queryText, params, isAcidTable, args);
   }
 
+  public List<Partition> getPartitionsByPartitionIdsInBatch(String catName, String dbName,
```

Comment:
nit: can we declare this method as `private`?

Reply:
Done.